src/client.rs

```diff
@@ -591,6 +594,58 @@ impl Client {
         }
     }
 
+    async fn on_invite(&self, message: Message) {
+        // TODO(diath): ERR_CHANOPRIVSNEEDED, RPL_AWAY.
+        if message.params.len() < 2 {
+            self.send_numeric_reply(
+                NumericReply::ErrNeedMoreParams,
+                "INVITE :Not enough parameters".to_string(),
+            )
+            .await;
+            return;
+        }
+
+        let nick = self.nick.lock().await.to_string();
```

Comment:
nick & user assignment can be moved lower, so they don't end up unused when branch exits 😎